/acquisition_left:
  ros__parameters:
    serialNumber: "KE0220040188"
    exposureTime: 1000
    autoExposure: False # Mporoun na mpoun kai True kai exposureTime, ta exw kanei handle sthn
    # klash ths cameras, ola komple
    expectedGrayValue: 50
    ROIx: 1280
    ROIy: 1024
    orientation: "left"


/acquisition_right:
  ros__parameters:
    serialNumber: "KE0220030138"
    exposureTime: 1000
    autoExposure: False
    expectedGrayValue: 50
    ROIx: 1280
    ROIy: 1024
    orientation: "right"


/slam:
  ros__parameters:
    velocity_estimation_frequency: 40

    # This sets the mode of operation. Set true for slam and false for localization
    mapping_mode: true
    track_map: ""

    # Defines the starting position of the car
    starting_position: [ -6.847, 0.0, 0.0 ] #before -7.5
    starting_position_covariance: [ 0.5, 0.1, 0.1 ]

    # The max distance perception measurements are accepted
    perception_range: 13.0

    # After how many time steps the optimization algorithm would be executed
    optimization_interval: 4

    # The max distance at which two cones would be associated
    association_threshold: 1.9
    min_observations: 4

    # iSAM parameters
    relinearize_threshold: 0.1
    relinearize_skip: 1

    # Big orange cones for starting/finish line
    left_orange: [0.0, -2.5]
    right_orange: [0.0, 2.5]

    lap_counter_cooldown: 10

    logger: true


/saltas:
  ros__parameters:
      velocity_estimation_frequency: 40   # Hz
      perception_frequency: 10            # Hz


/velocity_estimation:
  ros__parameters:
    frequency: 40                         # [Hz]
    initial_state_vector: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0]

    initial_state_covariance: [1.0e-3,    0.0,    0.0,    0.0,    0.0,    0.0,
                                  0.0, 1.0e-3,    0.0,    0.0,    0.0,    0.0,
                                  0.0,    0.0, 1.0e-3,    0.0,    0.0,    0.0,
                                  0.0,    0.0,    0.0, 1.0e-3,    0.0,    0.0,
                                  0.0,    0.0,    0.0,    0.0, 1.0e-3,    0.0,
                                  0.0,    0.0,    0.0,    0.0,    0.0, 1.0e-3]
    
    process_noise_covariance: [7.0e-3,    0.0,    0.0,    0.0,    0.0,    0.0,
                                  0.0, 8.0e-3,    0.0,    0.0,    0.0,    0.0,
                                  0.0,    0.0, 5.0e-6,    0.0,    0.0,    0.0,
                                  0.0,    0.0,    0.0, 5.0e-4,    0.0,    0.0,
                                  0.0,    0.0,    0.0,    0.0, 3.0e-4,    0.0,
                                  0.0,    0.0,    0.0,    0.0,    0.0, 1.0e-4]

    measurement_noise_covariance: [   0.030,-13.0e-4,      0.0,       0.0,       0.0,  0.0,   0.0,
                                   -13.0e-4,   0.024,      0.0,       0.0,       0.0,  0.0,   0.0,
                                        0.0,     0.0, 1.22e-06,       0.0,       0.0,  0.0,   0.0,
                                        0.0,     0.0,      0.0,     0.032,-7.612e-05,  0.0,   0.0,
                                        0.0,     0.0,      0.0,-7.612e-05,     0.051,  0.0,   0.0,
                                        0.0,     0.0,      0.0,       0.0,       0.0, 60.0,   0.0,
                                        0.0,     0.0,      0.0,       0.0,       0.0,  0.0, 180.0]

    # Sensor Positions
    vectornav-vn-200:
      x: 0.49                   # [m]
      y: 0.00                   # [m]
      roll: -0.98              # [deg]
      pitch: -4.53             # [deg]
    vectornav-vn-300:
      x: -0.5262                # [m]
      y: 0.00                   # [m]
      roll: 0.00                # [deg]
      pitch: 10.00              # [deg]
    
    # Car characteristics
    wheel_radius: 0.1995        # [m]
    front_axle: 0.76            # [m]
    rear_axle: 0.89             # [m]


/path_planning:
  ros__parameters:
    #triangulation
    maximum_angle: 90
    maximum_edge_angle: 60
    maximum_distance: 5
    target_depth: 12

    #penalties
    same_edge_penalty: 100
    length_penalty: 0.1
    angle_penalty: 0.1
    total_length_reward: 0.5
    filtering_threshold: 90

    #local map selection
    selection_radius_small: 7
    selection_radius_big: 15
    selection_angle: 45


/pure_pursuit:
  ros__parameters:
    discipline: "Autocross"
    #model
    mass: 193.5
    gravitational_acceleration: 9.81
    wheelbase: 1.59
    weight_distribution: 0.467
    h_cog: 0.27
    air_density: 1.225
    Cd*A: 2.0
    Cl*A: 7.0
    gear_ratio: 3.9
    wheel_radius: 0.2054
    efficiency: 0.85
    Fz_0: 1112.0554070627252
    max_positive_torque: 186.2              #EDIT BEFORE
    max_negative_torque: -186.2             #EDIT BEFORE
    minimum_weight_distribution_rear: 1.0  
    c_tire: 0.66

    #Pure Pursuit
    ld_min: 2.5                             #EDIT BEFORE
    ld_max: 7.0                             #EDIT BEFORE
    v_min: 0.0                              #EDIT BEFORE
    v_max: 10.0                             #EDIT BEFORE
    emergency_factor: 0.75
    emergency_threshold: 0.75
    safety_factor: 0.75
    max_speed: 5.0                         #EDIT BEFORE
    dt: 0.025
    safe_speed_to_break: 3.0

    #PID
    kp: 1000
    ki: 1000
    kd: 100
    PID_max_output: 3000
    Integral_max_output: 2000
    spline_resolution_per_meter: 10
    total_laps: 2                           #CHECK BEFORE

/mpc:
  ros__parameters:
    simulation: false
    mission: "autox"
    total_laps: 2
    v_limit: 4.0
    node_freq: 40
    s_interval: 0.15
    distance_safe: 0.9
    emergency_forward: 1.0
    F_init: 300.0
    F_max: 500.0
    F_min: -500.0
    s_space_max: 0.5
    s_space_min: 0.1  